Add integration test for CPU and Platform profiles

officialcjunior opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
* Add an integration test for CPU and Platform profiles.

* Return `NULL` if `target->platforms` is `NULL` in
`rz_arch_platform_target_new`

* Move the call of the API which adds the flags/comments for both
CPU and platform profiles to `rz_core_analysis_all()`.

**Test plan**

- [x] CI is green

**Closing issues**

None

officialcjunior committed:
Add integration test and a fix coverity scan issue

* Return `NULL` if `target->platforms` is `NULL` in
`rz_arch_platform_target_new`

* Move the call of the API which adds the flags/comments for both
CPU and platform profiles inside `rz_core_analysis_all()`.

officialcjunior commented:
Right. The test I just introduced runs fine, and the other fails because the directory `.tmp/` where it stores the `.rzdb` is already there, I believe. 
Should I remove that check (at line 40) or get another directory?

XVilka commented:
But it should be fine even if it exists, I think. See also related https://github.com/rizinorg/rizin/issues/1093

ret2libc commented on test/integration/test_cpu_platform_profiles.c:
```diff
@@ -0,0 +1,124 @@
+#include "../unit/minunit.h"
+
+bool test_cpu_profiles() {
+
```
```suggestion
```

ret2libc commented on test/integration/test_cpu_platform_profiles.c:
```diff
@@ -0,0 +1,124 @@
+}
+
+bool test_platform_profiles() {
+
```
```suggestion
```

officialcjunior replied:
Thank you!

ret2libc commented on test/integration/test_cpu_platform_profiles.c:
```diff
@@ -0,0 +1,124 @@
+#include <rz_core.h>
```
@thestr4ng3r I probably missed the addition of this dir. How are we going to split things that need to go in test/integration and test/unit? I believe there are already some things in unit that are not that "unit". WDYT?

XVilka replied:
Integration tests in this case are the ones that involve multiple parts/modules of Rizin, and usually represent some real use-case scenario.  They are good way to test possible use of the Rizin API.

officialcjunior committed:
Suggestions from @ret2libc

Remove unwanted spaces.

Co-authored-by: Riccardo Schirone <ret2libc@users.noreply.github.com>

officialcjunior committed:
Use `rz_file_is_directory` instead of `rz_file_exists` for the
integration tests

This condition is while checking whether a directory already exists
for storing the Project database file.

wargio commented on test/integration/test_open_analyse_save_load_project.c:
```diff
@@ -37,7 +37,7 @@ bool test_open_analyse_save() {
 
 	// 4. Save into the project
-	if (!rz_file_exists(".tmp/")) {
+	if (!rz_file_is_directory(".tmp/")) {
```
is this folder even legal for windows? @ret2libc

XVilka replied:
It is - test was passing on Windows

ret2libc replied:
Yeah I think it is nowadays.. anyway, fixed on another PR.

officialcjunior committed:
Use `RZ_SYS_DIR` on the path while checking for `.tmp` folder

wargio commented on test/integration/test_open_analyse_save_load_project.c:
```diff
@@ -37,7 +37,7 @@ bool test_open_analyse_save() {
 
 	// 4. Save into the project
-	if (!rz_file_exists(".tmp/")) {
+	if (!rz_file_is_directory(".tmp/")) {
```
You can remove this commit, since ret2libc has fixed this on another PR which is merged

officialcjunior replied:
Rebased to `dev`.

ret2libc commented on test/integration/meson.build:
```diff
@@ -1,7 +1,8 @@
     'open_analyse_save_load_project',
-    'bin_vfiles'
+    'bin_vfiles',
+    'cpu_platform_profiles'
```
Not important, but I think this should be valid in meson.build files. This way you don't have to modify the last line whenever you add a new entry to the array
```suggestion
    'cpu_platform_profiles',
```

ret2libc reviewed (approved):
Seems ok for what I understand.

On a different topic: I like the split in integration/unit tests! We could use the meson feature `suite: '` of `test` to split them and then run only unit by default.

officialcjunior committed:
Add a comma at the end of the list on \`meson.build\`

Add a comma so that we don't have to modify the last line whenever you add a new entry to the array.

Co-authored-by: Riccardo Schirone <ret2libc@users.noreply.github.com>
